sdk/monitor/monitor-opentelemetry/src/traces/sampler.ts

```diff
@@ -19,8 +19,12 @@ export class ApplicationInsightsSampler implements Sampler {
    */
   constructor(samplingRatio: number = 1) {
     this._samplingRatio = samplingRatio;
-    if (this._samplingRatio > 1) {
-      throw new Error("Invalid sampling rate, sampling rate must be a value in the range [0,1].");
+    if (
+      this._samplingRatio > 1 ||
+      this._samplingRatio < 0 ||
+      !Number.isFinite(this._samplingRatio)
```

Comment:
I'm not sure if this code is actually reachable here, NaN and other infinite values should return false in other checks I will assume

Reply:
It is currently as the config file only checks that the value is undefined (matching the behavior of the jsonConfig check). I've considered moving this logic to the `config` file, but I believe it's safest and avoids code duplication to check on the initialization of the Sampler itself such that no matter what method we use to configure, this check will always catch an invalid value.